test/org/impact2585/frc2016/tests/InputMethodTest.java

```diff
@@ -0,0 +1,251 @@
+ */
+public class InputMethodTest {
+	private TestInput input;
+	
+	private double forwardMovement;
+	private double rotationValue;
+	private boolean invert;
+	
+	private double analogTopArm;
+	private boolean digitalTopArmForward;
+	private boolean digitalTopArmBackward;
+	
+	private double bottomArmForward;
+	private double bottomArmBackwards;
+	private double reversibleBottomArmValue;
+	
+	private boolean intake;
+	private boolean outake;
+	
+	private double analogMoveIntakeTowardsBot;
+	private double analogMoveIntakeAwayFromBot;
+	private boolean digitalMoveIntakeTowardsBot;
+	private boolean digitalMoveIntakeAwayFromBot;
+	
+	/**
+	 * sets up the test
+	 */
+	@Before
+	public void setUp() {
+		input = new TestInput();
+		invert = false;
+		digitalTopArmForward = false;
+		digitalTopArmBackward = false;
+		outake = false;
+		intake = false;
+		digitalMoveIntakeAwayFromBot = false;
+		digitalMoveIntakeAwayFromBot = false;
+	}
+
+	/**
+	 * test cases for the TestInput
+	 */
+	@Test
+	public void test() {
+		//tests the drivetrain
+		forwardMovement = 1;
+		Assert.assertTrue(input.forwardMovement() == 1);
+		
+		//tests moving the top arm with analog input
+		analogTopArm = 1;
+		Assert.assertTrue(input.moveTopArm() == 1);
+		
+		//tests moving the top arm with digital input
+		digitalTopArmForward = true;
+		analogTopArm = 0;
+		Assert.assertTrue(input.moveTopArm() == 1);
+		
+		//tests if the top arm doesn't move if Harris presses both digital input buttons
+		digitalTopArmBackward = true;
+		Assert.assertTrue(input.moveTopArm() == 0);
+		
+		//tests if the top arm can move backwards with digital input
+		digitalTopArmForward = false;
+		Assert.assertTrue(input.moveTopArm() == -1);
+		
+		//tests that the bottom arm moves with two analog buttons
+		bottomArmForward = 1;
+		Assert.assertTrue(input.moveBottomArm() == 1);
+		
+		//tests if the bottom arm doesn't move if Harris presses both triggers
+		bottomArmBackwards = 1;
+		Assert.assertTrue(input.moveBottomArm() == 0);
+		
+		//tests if the bottom arm moves backwards
+		bottomArmForward = 0;
+		Assert.assertTrue(input.moveBottomArm() == -1);
+		
+		//tests if the bottom arm moves with a reversible analog input
+		bottomArmBackwards = 0;
+		reversibleBottomArmValue = -1;
+		Assert.assertTrue(input.moveBottomArm() == -1);
+		
+		//tests if the input arms move away with analog input
+		analogMoveIntakeAwayFromBot = 1;
+		Assert.assertTrue(input.moveIntake() == -1);
+		
+		//tests if the input arms move towards the bot with analog input
+		analogMoveIntakeAwayFromBot = 0;
+		analogMoveIntakeTowardsBot = 1;
+		Assert.assertTrue(input.moveIntake() == 1);
+		
+		//tests if the input arms don't move if Harris presses both triggers
+		analogMoveIntakeAwayFromBot = 1;
+		Assert.assertTrue(input.moveIntake() == 0);
+		
+		//tests if the input arms don't move if one of the analog inputs are negative
+		analogMoveIntakeAwayFromBot = -1;
+		Assert.assertTrue(input.moveIntake() == 0);
```

Comment:
That's interesting that you're testing for that. If one of the triggers gave a negative value, that would mean that something has gone horribly wrong.